### PR TITLE
SqlClientFactory: Override CreateDataAdapter

### LINKED ISSUE #19651: SqlClientFactory.CreateDataAdapter returns null
I'm running the .NET Core 2.0 bits and testing against some legacy full framework packages (yeah some legacy code). Looks like most of the other services are working, just the data adapter isn't.

image

Is this something that will come later?

### PR DESCRIPTION
**Pull request by justinvp**
Override `CreateDataAdapter` and add tests.

Fixes #19651, #19673

cc: @saurabh500, @corivera
fyi: @RickStrahl, @FransBouma

### CODE REVIEW

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlClientFactoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        public void CreateDataAdapter_NotNull()
+        {
+            Assert.NotNull(SqlClientFactory.Instance.CreateDataAdapter());
+        }
+
+        [Fact]
+        public void CreateParameter_NotNull()
+        {
+            Assert.NotNull(SqlClientFactory.Instance.CreateParameter());
+        }
```
Comment (stephentoub) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlClientFactoryTest.cs]: Is there nothing more stringent we can check, like the type of the thing that comes back?

**Comment by justinvp** [comment]
@dotnet-bot test Innerloop Tizen armel Debug Cross Build please

**Comment by ViktorHofer** [comment]
@justinvp thanks for the quick fix!

**Comment by danmosemsft** [comment]
@justinvp thanks - take a look here if you want to do more ! -- https://github.com/dotnet/corefx/issues/17126

**Pull request merged**